src/docker/restorableappengine.cc

```diff
@@ -299,6 +319,87 @@ void RestorableAppEngine::pullApp(const Uri& uri, const boost::filesystem::path&
        boost::process::start_dir = app_dir);
 }
 
+void RestorableAppEngine::checkAppUpdateSize(const Uri& uri, const boost::filesystem::path& app_dir) const {
+  const Manifest manifest{Utils::parseJSONFile(app_dir / Manifest::Filename)};
+  const Json::Value engine_info{docker_client_->engineInfo()};
+  if (!(engine_info.isObject() && engine_info.isMember("Arch"))) {
+    LOG_WARNING << "Failed to get an info about a system architecture";
+    return;
+  }
+
+  const auto arch{engine_info["Arch"].asString()};
+  const auto layers_manifest{manifest.layersManifest(arch)};
+  if (!layers_manifest.isObject()) {
+    LOG_WARNING << "App layers' manifest is missing, skip checking an App update size";
```

Comment:
once compose-publish is updated, this would be a real error, right? i.e. maybe LOG_ERROR here?

Reply:
Not sure. Even after compose-publish is updated there can be Targets that are not installed on devices and such that refer to Apps published by older version of compose-publish. IMO, it's quite valid use-case?

Reply:
i guess its something we can adjust later once people are all updated.